### ISSUE TITLE
Reject text/* content-type responses as dead streams

### ISSUE THREAD

**Issue by kristofferR** [NewsGuyTor/IPTVChecker#41]
## Problem

When a server returns an HTML error page (e.g., "403 Forbidden" body with 200 status, or a captive portal), the Content-Type is typically `text/html` or `text/plain`. The Python version doesn't explicitly check for this, potentially treating HTML error pages as valid streams if they pass the data threshold.

The Rust version explicitly treats `text/*` responses as dead/error.

## Solution

Add a check in the `verify()` function after getting the content type. If the content type starts with `text/`, mark the stream as dead:

```python
content_type = resp.headers.get('Content-Type', '')
if content_type.lower().startswith('text/'):
    logging.debug(f"Text content-type response ({content_type}), marking as dead")
    return 'Dead', None
```

This should be added after the status code checks and before the playlist/direct-stream detection.

## Impact
Prevents false-positive "Alive" results for streams behind captive portals or returning HTML error pages with 200 status codes.
